core/src/main/java/com/microsoft/applicationinsights/internal/authentication/AzureMonitorRedirectPolicy.java

```diff
@@ -0,0 +1,71 @@
+package com.microsoft.applicationinsights.internal.authentication;
+
+// This is a copy from Azure Monitor Open Telemetry Exporter SDK AzureMonitorRedirectPolicy
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.net.HttpURLConnection;
+
+public final class AzureMonitorRedirectPolicy implements HttpPipelinePolicy {
+
+    private static final int PERMANENT_REDIRECT_STATUS_CODE = 308;
+    // Based on Stamp specific redirects design doc
+    private static final int MAX_REDIRECT_RETRIES = 10;
+    private final ClientLogger logger = new ClientLogger(AzureMonitorRedirectPolicy.class);
+    private String redirectedEndpointUrl;
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        return attemptRetry(context, next, context.getHttpRequest(), 0);
+    }
+
+    /**
+     *  Function to process through the HTTP Response received in the pipeline
+     *  and retry sending the request with new redirect url.
+     */
+    private Mono<HttpResponse> attemptRetry(final HttpPipelineCallContext context,
+                                            final HttpPipelineNextPolicy next,
+                                            final HttpRequest originalHttpRequest,
+                                            final int retryCount) {
+        // make sure the context is not modified during retry, except for the URL
+        context.setHttpRequest(originalHttpRequest.copy());
```

Comment:
does this need to pass around `originalHttpRequest` or can it use `context.getHttpRequest()` each time?

Reply:
I believe we can use context.getHttpRequest, but I followed the same standard of setting the context from https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java#L96